### PR TITLE
[EPT-1715] new page modal 1

### PR DESCRIPTION
**Pull request by shanthru**
adding a modal to create page

### CODE REVIEW

File: src/ListView/ListView.jsx [hunk 1]
```diff
@@ -38,11 +39,13 @@ function ListView() {
...
   return (
-    <PageTable
-      loading={loading}
-      pages={Object.values(pageMap)}
-      getPageById={getPageById}
-    />
+    <ListViewLayout>
+      <PageTable
+        loading={loading}
+        pages={Object.values(pageMap)}
```
Comment (jdrury-sailthru) [inline src/ListView/ListView.jsx:45]: we should maybe wrap a useMemo around this `Object.values` to prevent unnecessary rerenders

it would have to be done above this render statement

File: src/ListView/ListViewLayout.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  return (
+    <>
+      <ListLayout
+        pageTitle="Hosted Pages"
+        cta={
+          <Button
+            onClick={() => {
+              setModalTitle(() => "New HTML Page");
+              setMode(() => "html");
```
Comment (jdrury-sailthru) [inline src/ListView/ListViewLayout.jsx:20]: ugh this is smelly

File: src/ListView/ListViewLayout.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const [mode, setMode] = useState();
+
+  return (
+    <>
+      <ListLayout
+        pageTitle="Hosted Pages"
+        cta={
+          <Button
+            onClick={() => {
+              setModalTitle(() => "New HTML Page");
```
Comment (jdrury-sailthru) [inline src/ListView/ListViewLayout.jsx:19]: nit ideally this would be an internal concern of the modal

File: src/ListView/features/CreatePage/CreatePageModal.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      .then(done)
+      .catch((e) => setError(true));
+    //             .finally(setDisplayModal((state) => !state))
+  }
+  useEffect(() => {
+    setDisableSubmitButton(false);
+    if (!state.name || !state.type) {
+      setDisableSubmitButton(true);
+    }
+  });
```
Comment (jdrury-sailthru) [inline src/ListView/features/CreatePage/CreatePageModal.jsx:26]: why is this necessary? what problem does it solve

Comment (jdrury-sailthru) [reply src/ListView/features/CreatePage/CreatePageModal.jsx:26]: this seems wrong because useEffect is an escape hatch so that react can be made aware of things outside the context of react

eg network calls, dom events

but these changes seem to be well within the bounds of react

File: src/ListView/features/CreatePage/CreatePageModal.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const [error, setError] = useState(false);
+
+  function handleSubmit(done) {
+    //FIXME
+    setDisableSubmitButton(true);
+    pagesApi
+      .createPage({ ...state, mode })
+      .then(done)
+      .catch((e) => setError(true));
+    //             .finally(setDisplayModal((state) => !state))
```
Comment (jdrury-sailthru) [inline src/ListView/features/CreatePage/CreatePageModal.jsx:19]: probably also should add 
```
.finally(() => setDisableSubmitButton(false));
```

File: src/ListView/features/CreatePage/CreatePageModal.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  const [disableSubmitButton, setDisableSubmitButton] = useState(false);
+  const [error, setError] = useState(false);
+
+  function handleSubmit(done) {
+    //FIXME
+    setDisableSubmitButton(true);
+    pagesApi
+      .createPage({ ...state, mode })
+      .then(done)
+      .catch((e) => setError(true));
```
Comment (jdrury-sailthru) [inline src/ListView/features/CreatePage/CreatePageModal.jsx:18]: see SMS campaign builder (the new one) for example error handling UX 

the schedule confirmation modal shows the error in a banner at the top of the modal

perhaps @andoArbST can show you

**Comment by shanthru** [comment]
closing in favor of https://github.com/sailthru/hosted-pages-frontend/pull/10 and https://github.com/sailthru/hosted-pages-frontend/pull/11
